pkg/api/validation/validation.go

```diff
@@ -45,6 +45,7 @@ const isNegativeErrorMsg string = `must be non-negative`
 const fieldImmutableErrorMsg string = `field is immutable`
 const cIdentifierErrorMsg string = `must be a C identifier (matching regex ` + validation.CIdentifierFmt + `): e.g. "my_name" or "MyName"`
 const isNotIntegerErrorMsg string = `must be an integer`
+const headlessServicePortErrorMsg string = `headless service port must equal to targetPort`
```

Comment:
actually, the while string should be "must be equal to targetPort when clusterIP = None"